apps/src/aichat/views/ChatWorkspace.tsx

```diff
@@ -1,35 +1,96 @@
-import React from 'react';
+import React, {useState} from 'react';
 import PanelContainer from '@cdo/apps/lab2/views/components/PanelContainer';
 import ChatMessage from './ChatMessage';
 import UserChatMessageEditor from './UserChatMessageEditor';
 import moduleStyles from './chatWorkspace.module.scss';
 import aichatI18n from '../locale';
-
+import {ChatCompletionMessage} from '../types';
+import {Role, Status} from '../constants';
 import {demoChatMessages} from './demoMessages'; // demo chat messages - remove when connected to backend
+import {openaiCompletion} from '../openai';
 
 const ChatWorkspace: React.FunctionComponent = () => {
+  const [storedMessages, setStoredMessages] =
+    useState<ChatCompletionMessage[]>(demoChatMessages);
+
+  const onSubmit = async (message: string) => {
```

Comment:
Also, just thinking ahead - this logic actually may be a good candidate for using a [redux async thunk](https://redux-toolkit.js.org/api/createAsyncThunk) once we have our redux store set up. We're changing some state after the submit button is clicked (creating a new 'unknown' chat message) and would probably need to convey some loading information while load is in progress. Doing this in an async thunk would allow us to keep that logic close to redux and simplify the component code. But this is definitely something we can look into later.

Reply:
Okay - that makes sense. @bencodeorg has a branch which is adding a Redux store fo aichat [in this branch.](https://github.com/code-dot-org/code-dot-org/compare/staging...ben/openai-fe-client) I'll connect with him when he's back about this idea - thanks!